### ISSUE TITLE
Incorrect drag after adding pane programatically

### ISSUE THREAD

**Issue by dadebue** [svecosystem/paneforge#28]
### Describe the bug

Dragging a PaneResizer after inserting an pane programatically leads to incorrect size changes when the pane was added with splice or unshift. When pushing a pane to the array everything is fine.

https://github.com/svecosystem/paneforge/assets/81422785/cd47df6d-c41a-47fc-bf21-d00f93e03e03

### Reproduction

- create layout with multiple groups
- add a pane programatically with splice or unshift

REPL: https://svelte-5-preview.vercel.app/#H4sIAAAAAAAAE8VVXa_jNBD9K7NepLRSPm5LEZBtCggktA9IK3i8uQ9pMklMHduynXtvN-S_I9tJmsIi9glUpY1nMmfOnMxMB1JThpqkjwPhRYckJT9ISUJirtIe9DMygyQkWvSqtJajLhWVBljBmywnRufklHMA2kmhDAzwoeD4sxK9DN2t__4VNf2ICkaoleggkAXHWqgGg3c5z80UvIT-KDopOHIzPR4niyv2lFwczw1YovDL1fohg8HZgFYpaKMob955Q0UVloYKnkLQCkU_Cm4KFsAfEDyjMrT0h55XWFOO1RRmaep0gn98ctbRfduLoYHGckpvBL7QpjC4GawknkewC0J_-gcSk3fK9ehPAIOP3gcwhovNpXYf5_zSBs-WNfxS1Mo_J5iRDxZ5PnwVwAhPS6I5pbeM20ntUnBtQPCS0fICGWy2kJ1mVk6K2GV53D35m1hLRkvc7MKHcMr0dQDj1gk563iHuvssWNnrdjMBfrMCPCa-PU-W79G3StpiUZ1yfmSUX0Ahy3KizZWhbhFNTqBVWGc5aY2ROk2SnstLE5eiS0xB2QvlVan19_t4H---TSqqzWKPO8rjUuucJCeX-y5fzo_n3hjBoWSF1llOOhMdoGPRAc5CVahARrucwDCVPp7ec43KADXYwcZLtz0mHuX02XgTXDbj7sbTh163DnaNVtHnBUpGh5ycfrMpKW9gIxT0XLe0NpQ3WzgrLC4aKlU0jfWXBSt7VthuA1GDaRFuHQ3KT7t-c0wq-uylWCdTaEOfEdqo7hmDF_8jX6MDyHO0P0w75fiJhXCmvEpdT4AX3WcgIelERWuKFUmN6nEMl422oPzne81e_9uCArALavAryq4mK11xZrjZwpgOd5trtH6phNQbO-v3Y7RUfmOZ-fB4MYxAq9lIq3F51fdvWAlbQBWxZlJ3eItF2cJqxqHQIEOg1evoV5BL79Ad8mnZhW9pDWsGswNgHnyNrPbY2SBH1y9TbIpM4zpi3Z4T2Yl6zfDVjY6OSuQGFfzea0Pr63y8FTWP4bmJGlVco93Dg6tz3sC58TLedIoHW2gK738CX97t4bmzJ8YJrWdB3J_h5HEq0OoV3mTZWsaYIW9MCxHsljKP64792yz-a53nJjoX5aVx9cJfGyDLslunfhe00T5Ig5doH4w5WZRfqhgS-97dwl6665NT_DT-CZWTrjejCAAA

### System Info

```bash
svelte-5-preview vercel
```

### Severity

annoyance

**Comment by huntabyte** [comment]
Are you able to reproduce this in Svelte 4?

**Comment by dadebue** [comment]
Yes: https://svelte.dev/repl/c89d1fc9ebc5415a9e890388f279536e?version=4.2.18

**Comment by huntabyte** [comment]
Okay here's a workaround at least for the time being: https://svelte.dev/repl/c89d1fc9ebc5415a9e890388f279536e?version=4.2.18

When I find some time I'll try to investigate why it is getting registered backwards like that.

**Comment by dadebue** [comment]
Did you mean to paste a different link? The link you provided is the REPL I created so I still get the problem 🙈

**Comment by huntabyte** [comment]
Whoops, here you go [REPL](https://svelte.dev/repl/1077a4e80bbf47fd8335f1255878f84e?version=4.2.18)

**Comment by LePau** [comment]
I had the same issue with [collapsing a panel using a key block](https://github.com/svecosystem/paneforge/issues/41#issuecomment-2554913639), but that doesn't seem to fix the drag issue in that case.  Any ideas?

Just adding here as another data point.

**Comment by huntabyte** [comment]
I'm unsure how this could be solved by paneforge, as I don't see a way for us to know what pattern you are using on the outside to mutate arrays. Even adding a unique key to the `{#each`  still seems to produce the opposite drag behavior, which leads me to believe something is getting confused.

I'll look into this further when I have the time, if someone else wants to put a second set of eyes it would be appreciated.

**Comment by rdrpenguin04** [comment]
Still getting issues related to this; I'm considering taking a look into the bug because I'd rather not write my own resizer system when this _should_ work just fine. Is there any documentation on how the internal state gets created and managed that I could look at?

**Comment by rdrpenguin04** [comment]
Nevermind; all that was needed to fix my case was to use `order`. None of the directions are reversed; everything "just works".

**Comment by adamdva** [comment]
I was having the same on a simple horizontal left side panel and main panel.
Adding order to both panels fixed it.
